cataclysmbnteam/Cataclysm-BN
# Move aftershock cbms to base game

Move aftershock cbms to base game



#### Summary
SUMMARY: Content "Move aftershock cbms to base game"


#### Purpose of change
Aftershock mods contains intresting cmbs. It is good idea to add them to base game, because:
1) Some of cbms was intentionally moved to aftershock from DDA, because they stopped fitting DDA new lore. BN lore supports this level of technology.
2) This cbms already has code support but without Aftershock it is dead code.

List of added CBMs:
1) Solar Panels CBM: "Installed on your back is a set of retractable solar panels.  When in direct sunlight, they will automatically deploy and slowly recharge your power level.". It was moved out of DDA because it was deemed unrealistic. Previously was preinstalledon bionic sniper profession. Should be still good power generation option.
3) Deployable Grenade Launcher CBM:"Your right hand can fold inward to reveal a mechanism capable of igniting and firing 40mm grenades across medium distances.". Interesting option for player that want heavy weapon at hand. Buffed: increased accuracy, now can be preloaded with 3 grenades. Unique role: almost energy independent bionic heavy weapon. Good accuray.
4) Linguistic Coprocessor CBM: "The left hemisphere of your brain has been augmented with a microcomputer that moderately increases the speed that language and written words are processed, granting a 15% increase to reading speed." I don't know why it is not base game. Increases reading speed by 15%. Not 25% for some reason.
5) Dopamine Stimulators CBM: Tiny cybernetic stimulators have been installed throughout your ventral tegmental area, and trigger at set intervals using bionic power.  This slowly releases a stream of reward chemicals and hormones into your brain, inducing a state of euphoria that notably elevates mood.". Kinda built in vibrator analog.
6) Fusion Blaster Arm CBM: "Your left arm has been replaced by a heavy-duty fusion blaster!  You may use your energy banks to fire a damaging heat ray.  However, you are unable to use or carry two-handed items, and your strength limits what you can use with your one hand.". No longer remove hand. Armour piercing buffed significantly. Projectile damage type changed to heat. Unique role: Bionic heave energy weapon.

Obviously cbm re balance will affect Aftershock mod as well due the way things are coded, but it should benefiting.

List of  not added CBMs:
1) Skullgun CBM: "Concealed in your head is a single shot .40 pistol.  Activate the bionic to fire and reload the skullgun.". Built in .40 pistol. Reduces intellect and perception **It is joke gun mostly**.
2)   Cranium Bomb: "You've worked for some nasty people.  People who installed a bomb at the top of your spine.  They are all dead now but there is unfortunately a dead man switch if you don't check in roughly every thirty days.  You need this out and fast.". **Scenario based thing only. Preinstalled in Bionic Operator Aftershock profession. Should be ported  only with profession**
3) Wind Turbine CBM: "Installed on your body is a set of small retractable wind turbines.  When activated, they will deploy and slowly harvest wind power to recharge your power level.". Wind dependent power generation. I**t mostly duplicates what solar does, but more randomly. Wind isn't communicated to player well, it's hard to see it and even harder to predict it.
The bionic also kills immersion by being obviously weird in a way that can't be explained with "it's magic sci-fi stuff". It's like a propeller hat, except it makes enough power to run bionics off it.**




#### Describe the solution
Merge almost all CBMs from Afershock mod to base game.


#### Describe alternatives you've considered
None.


#### Testing
1) Check if Aftershock CBMs funtional.
2) Check if Aftershock CBMs at least adequate balance.


- Solar Panels CBM - working. Reacting on sunlight.
- Deployable Grenade Launcher CBM - it is working. Got exception on shooting explosive but not slug. **Never mind m79 grenade launcher crashes as well on explosion.** It seems CMB is working, but 40x46 and 40x53 ammo need checking. It is actually less accurate than m79, so it is much description. I assume it is good compromise.
- Dopamine Stimulators CBM- +20 mood buff. 5Kj each 200 turns.  5Kj activation. Helps maintain focus at the cost of energy. No side effects. I see potential uses in long tasks. Looking good. Moved to military CBM spawn list for balance.
- Fusion Blaster Arm CBM: No longer removes hands, just integrates with it. Damage changed to heat. Armour piercing significantly increased.
- Linguistic Coprocessor CBM: buffed. Now it increases reading speed to +25% instead of 15%.

**Note that balance chages does affect Aftershock users, but It supposed to be in good way.**
#### Additional context
**Save for testing(afscbm):**
[afscbm.zip](https://github.com/cataclysmbnteam/Cataclysm-BN/files/9276638/afscbm.zip)

**Save for testing with Aftershock mod (afscmbs, s at the end of save dir name):**
[afscmbs.zip](https://github.com/cataclysmbnteam/Cataclysm-BN/files/9353381/afscmbs.zip)
